test.cpp

```diff
@@ -920,10 +921,87 @@ static void ifA()
 
 static void ifCharLiteral()
 {
-    const char code[] = "#if ('A'==0x41)\n"
+    const char code1[] = "#if ('A'==0x41)\n"
                         "123\n"
                         "#endif";
-    ASSERT_EQUALS("\n123", preprocess(code));
+    ASSERT_EQUALS("\n123", preprocess(code1));
+    
+    const char code2[] = "#if ('\\''==0x27)\n"
```

Comment:
hmm.. this is not how I meant.

Reply:
Sorry, I might have misunderstood. Do you want all tests combined into one `ASSERT_EQUALS`?

Reply:
I think it seems like a good idea to have a single `ifCharLiteral` test to test that character literals are evaluated..

but then I want to have separate unit test(s) for characterLiteral. They should not preprocess just call characterLiteral:

    void characterLiteral() {
         ASSERT_EQUALS((int)'\n', characterLiteralToLL('\n'));
         .... all various cases we want to test ....
    }